hera_cal/io.py

```diff
@@ -243,15 +247,131 @@ def load_cal(input_cal, return_meta=False):
+        times : type=ndarray, holds unique times of integration centers in Julian Date
+        pols : type=ndarray, holds unique polarization of gains in 'x' and/or 'y' form
+        flags : type=dictionary, holds boolean flags (True if flagged) for gains.
+                Must match shape of gains.
+        quality : type=dictionary, holds "quality" of calibration solution. Must match
+                  shape of gains. See pyuvdata.UVCal doc for more details.
+        write_file : type=bool, if True, write UVCal to calfits file
+        return_uvc : type=bool, if True, return UVCal object
+        outdir : type=str, output file directory
+        overwrite : type=bool, if True overwrite output files
+        gain_convention : type=str, gain solutions formatted such that they 'multiply' into data
+                          to get model, or 'divide' into data to get model
+                          options=['multiply', 'divide']
+        history : type=str, history string for UVCal object.
+        x_orientation : type=str, orientation of X dipole, options=['east', 'north']
+        telescope_name : type=str, name of telescope
+        cal_style : type=str, style of calibration solutions, options=['redundant', 'sky']. If
+                    cal_style == sky, additional params are required. See pyuvdata.UVCal doc.
+        kwargs : additional atrributes to set in pyuvdata.UVCal
+    Returns:
+        if return_uvc: returns UVCal object
+        else: returns None
+    '''
+    # helpful dictionaries for antenna polarization of gains
+    str2pol = {'x': -5, 'y': -6}
+    pol2str = {-5: 'x', -6: 'y'}
+
+    # get antenna info
+    antenna_numbers = np.array(sorted(map(lambda k: k[0], gains.keys())), np.int)
+    antenna_names = np.array(map(lambda a: "ant{}".format(a), antenna_numbers))
+    Nants_data = len(antenna_numbers)
+    Nants_telescope = len(antenna_numbers)
+    ant_array = np.arange(Nants_data)
+
+    # get polarization info
+    jones_array = np.array(map(lambda p: str2pol[p], pols), np.int)
+    pol_array = np.array(map(lambda j: pol2str[j], jones_array))
+    Njones = len(jones_array)
+
+    # get time info
+    time_array = np.array(times, np.float)
+    Ntimes = len(time_array)
+    time_range = np.array([time_array.min(), time_array.max()], np.float)
+    integration_time = np.median(np.diff(time_array)) * 24. * 3600.
+
+    # get frequency info
+    freq_array = np.array(freqs, np.float)
+    Nfreqs = len(freq_array)
+    Nspws = 1
+    freq_array = freq_array[None, :]
+    spw_array = np.arange(Nspws)
+    channel_width = np.median(np.diff(freq_array))
+
+    # form gain, flags and qualities
+    gain_array = np.empty((Nants_data, Nspws, Nfreqs, Ntimes, Njones), np.complex)
+    flag_array = np.empty((Nants_data, Nspws, Nfreqs, Ntimes, Njones), np.bool)
+    quality_array = np.empty((Nants_data, Nspws, Nfreqs, Ntimes, Njones), np.float)
+    for i, p in enumerate(pol_array):
+        for j, a in enumerate(antenna_numbers):
```

Comment:
This should be a loop over ant_array. From pyuvdata docs, ant array is the index of the antenna in data arrays.

Reply:
but its an enumerate, so `j` is the index, which is being used to index into `gain_array`. this must be `antenna_numbers` because `a` is used to index into the `gains` dictionary. right?

Reply:
In this case it can be either antenna_numbers or ant_array, but from the docs, ant_array is the parameter you want to use to do this, not antenna_numbers. This is more for consistency than anything. My understanding is that antenna_numbers can be in any order and is just a way to label/name the antennas where ant_array is the array you look at for indexing into the data arrays. It's the same loop you have just replacing antenna_numbers with ant_array.

Reply:
oh wait, just realized that ant_array is just an array from 0, nants-1. Why?

Reply:
Oh wait, I misunderstood the docs. I would be happy if it was something like for j, a in zip(ant_array, antenna_numbers)

Reply:
but I think that is exactly the same as what I have here: `enumerate(antenna_numbers)`